rust/hyperlane-base/src/types/multisig.rs

```diff
@@ -114,7 +119,7 @@ impl MultisigCheckpointSyncer {
         // Sort in descending order. The n'th index will represent
         // the highest index for which we (supposedly) have (n+1) signed checkpoints
         latest_indices.sort_by(|a, b| b.cmp(a));
-        if let Some(&highest_quorum_index) = latest_indices.get(threshold - 1) {
+        if let Some(&highest_quorum_index) = latest_indices.get(threshold_weight as usize - 1) {
```

Comment:
when the threshold_weight is not an index I dont really understand this logic

Reply:
yes, thanks for pointing it out. Will file it under "CRIT"